server/bin/pbench-server-prep-shim-002.py

```diff
@@ -0,0 +1,327 @@
+                "{}: Error in copying .md5 file to Destination path.", config.TS
+            )
+            try:
+                os.remove(dest / tbmd5.name)
+            except FileNotFoundError:
+                logger.error(
+                    "{}: Warning: cleanup of copy failure failed itself.", config.TS
+                )
+            quarantine((errors / controller), logger, tb, tbmd5)
+            nerrs += 1
+            continue
+
+        # Next, mv the "large" tar ball to the destination. If the destination
+        # is on the same device, the move should be quick. If the destination is
+        # on a different device, the move will be a copy and delete, and will
+        # take a bit longer.  If it fails, the file will NOT be at the
+        # destination.
+        try:
+            shutil.move(str(tb), str(dest))
+        except Exception:
+            logger.error(
+                "{}: Error in moving tarball file to Destination path.", config.TS
+            )
+            try:
+                os.remove(dest / resultname)
+            except FileNotFoundError:
+                logger.error(
+                    "{}: Warning: cleanup of copy failure failed itself.", config.TS
+                )
+            quarantine((errors / controller), logger, tb, tbmd5)
+            nerrs += 1
+            continue
+
+        # Now that we have successfully moved the tar ball and its .md5 to the
+        # destination, we can remove the original .md5 file.
+        try:
+            os.remove(tbmd5)
+        except Exception:
+            logger.error(
+                "{}: Warning: cleanup of successful copy operation failed.", config.TS
+            )
+
+        try:
+            os.symlink((dest / resultname), (dest / "TODO" / resultname))
+        except Exception:
+            logger.error("{}: Error in creation of symlink.", config.TS)
+            # if we fail to make the link, we quarantine the (already moved)
+            # tarball and .md5.
+            quarantine(
+                (errors / controller), logger, (dest / tb), (dest / tbmd5),
+            )
+            nerrs += 1
+            continue
+
+        ntbs += 1
+
+        nstatus = f"{nstatus}{config.TS}: processed {tb}\n"
+        logger.info(f"{tb.name}: OK")
+
+    return Results(
+        nstatus=nstatus,
+        ntotal=ntotal,
+        ntbs=ntbs,
+        nquarantined=nquarantined,
+        ndups=ndups,
+        nerrs=nerrs,
+    )
+
+
+def main(cfg_name):
+    if not cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable or use --config <file> on the"
+            " command line",
+            file=sys.stderr,
+        )
+        return 2
+
+    try:
+        config = PbenchServerConfig(cfg_name)
+    except BadConfig as e:
+        print(f"{_NAME_}: {e} (config file {cfg_name})", file=sys.stderr)
+        return 1
+
+    archive = Path(config.ARCHIVE)
```

Comment:
Seems like `config.ARCHIVE` is already a `Path()` object (see https://github.com/distributed-system-analysis/pbench/blob/br69/lib/pbench/server/__init__.py#L59).  we can just assign it to `archive` for convenience.